proposals/2020-07-01_aws-kms-keyring-redesign/change.md

```diff
@@ -0,0 +1,327 @@
+of these smaller-scoped keyrings.
+The smaller-scoped AWS KMS keyrings become the basic building blocks that keyring-producing operations use to satisfy different customer use cases.
+
+From a customer standpoint, naming will be clearer.
+Customers interact with the keyring-producing operations and do not directly interact with the basic building blocks.
+However, more advanced customers can directly interact with the smaller-scoped keyrings
+and write their own keyring-producing operations.
+This gives advanced customers more flexibility and reduces the need to fork ESDK code in order to meet a specific use case.
+
+## Reference-level Explanation
+
+### AWS KMS symmetric keyring
+
+**Behavior**
+
+The AWS KMS symmetric keyring MUST...
+
+1. Be configured with a single string identifying an AWS KMS CMK
+   ([key name](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#key-names))
+   and MUST be configured with a single AWS SDK KMS service client (AWS SDK client) for that key name.
+2. Use the key name and AWS SDK client provided at initialization for encryption and decryption.
+3. Fail on decrypt if the encrypted data key’s provider info does not match the keyring’s configured key name.
+4. Provide the encrypted data key’s [provider info](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/structures.md#key-provider-information)
+   as part of the [AWS KMS Decrypt API](https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html) call.
+
+The AWS KMS symmetric keyring MAY...
+
+1. Be configured with a list of string [grant tokens](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#grant-tokens)
+   to be included in all AWS KMS calls.
+
+**Decryption Contract**
+
+1. The keyring MUST have access to an AWS KMS client that is configured with credentials for an AWS principal
+   that passes an [AWS KMS Decrypt API](https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html)
+   authorization check for the encrypted data key.
+2. The encrypted data key’s [provider info](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/structures.md#key-provider-information)
+   MUST match the keyring’s configured key name.
+3. The encryption context MUST exactly match the encryption context used on encrypt.
+
+**Initialization**
+
+On keyring initialization, an AWS KMS symmetric keyring MUST define the following:
+
+- Key Name
+- AWS SDK client
+
+On keyring initialization, an AWS KMS symmetric keyring MAY define the following:
+
+- Grant Tokens
+
+### AWS KMS symmetric region discovery keyring
+
+Customers MUST declare their intent to use the AWS KMS symmetric region discovery keyring instead of the AWS KMS symmetric keyring.
```

Comment:
I think we need to be more precise about "declare their intent" here. I think what we want is "this MUST be a different keyring" but that's a fuzzy statement. Perhaps just "the keyring-producing operations that produce this kind of keyring MUST be distinct from those that produce non-discovery keyrings"?

Reply:
I'm leaving this sentence as-is, but adding an additional sentence for further clarification.